src/authentication.rs

```diff
@@ -0,0 +1,82 @@
+use async_graphql::{Context, Error, Result};
+use axum::http::HeaderMap;
+use bson::Uuid;
+use serde::Deserialize;
+
+// Authenticate-User HTTP header.
+#[derive(Deserialize, Debug)]
+pub struct AuthenticateUserHeader {
+    id: Uuid,
+    roles: Vec<Role>,
+}
+
+// Extraction of AuthenticateUserHeader from HeaderMap.
+impl TryFrom<&HeaderMap> for AuthenticateUserHeader {
+    type Error = Error;
+
+    // Tries to extract the AuthenticateUserHeader from a HeaderMap.
+    //
+    // Returns a GraphQL Error if the extraction fails.
+    fn try_from(header_map: &HeaderMap) -> Result<Self, Self::Error> {
+        if let Some(authenticate_user_header_value) = header_map.get("Authenticate-User") {
```

Comment:
the correct header is `Authorized-User`
See https://github.com/MiSArch/gateway/blob/main/.meshrc.yaml#L14C10-L14C26